src/parquet/column/statistics-test.cc

```diff
@@ -0,0 +1,212 @@
+    ASSERT_EQ("", statistics.EncodedMax());
+  }
+
+  void TestMerge() {
+    int num_null[2];
+    random_numbers(2, 42, 0, 100, num_null);
+
+    TypedRowGroupStatistics<TestType> statistics1(schema_.get());
+    GenerateData(1000);
+    statistics1.Update(values_ptr_, values_.size() - num_null[0], num_null[0]);
+
+    TypedRowGroupStatistics<TestType> statistics2(schema_.get());
+    GenerateData(1000);
+    statistics2.Update(values_ptr_, values_.size() - num_null[1], num_null[1]);
+
+    TypedRowGroupStatistics<TestType> total(schema_.get());
+    total.Merge(statistics1);
+    total.Merge(statistics2);
+
+    ASSERT_EQ(num_null[0] + num_null[1], total.null_count());
+    ASSERT_EQ(values_.size() * 2 - num_null[0] - num_null[1], total.num_values());
+    ASSERT_EQ(total.min(), std::min(statistics1.min(), statistics2.min()));
+    ASSERT_EQ(total.max(), std::max(statistics1.max(), statistics2.max()));
+  }
+
+  using TypedStats = TypedRowGroupStatistics<TestType>;
+
+  void TestFullRoundtrip(int64_t num_values, int64_t null_count) {
+    auto node = PrimitiveNode::Make("column", Repetition::OPTIONAL, TestType::type_num,
+        LogicalType::NONE, FLBA_LENGTH);
+    auto gnode = std::static_pointer_cast<GroupNode>(
+        GroupNode::Make("schema", Repetition::REQUIRED, std::vector<NodePtr>({node})));
+
+    GenerateData(num_values);
+
+    // compute statistics for the whole batch
+    ColumnDescriptor descr(node, 1 /* max definition level */, 0);
+    TypedStats expected_stats(&descr);
+    expected_stats.Update(values_ptr_, num_values - null_count, null_count);
+
+    auto sink = std::make_shared<InMemoryOutputStream>();
+    std::shared_ptr<WriterProperties> writer_properties =
+        WriterProperties::Builder().statistics("schema.column")->build();
+    auto file_writer = ParquetFileWriter::Open(sink, gnode, writer_properties);
+    auto row_group_writer = file_writer->AppendRowGroup(num_values);
+    auto column_writer =
+        static_cast<TypedColumnWriter<TestType>*>(row_group_writer->NextColumn());
+    std::vector<int16_t> definition_levels(null_count, 0);
+    definition_levels.insert(definition_levels.end(), num_values - null_count, 1);
+    column_writer->WriteBatch(num_values, definition_levels.data(), nullptr, values_ptr_);
+    column_writer->Close();
+    row_group_writer->Close();
+    file_writer->Close();
+
+    auto buffer = sink->GetBuffer();
+    std::unique_ptr<RandomAccessSource> source(new BufferReader(buffer));
+    auto file_reader = ParquetFileReader::Open(std::move(source));
+    auto rg_reader = file_reader->RowGroup(0);
+    auto stats = std::static_pointer_cast<TypedStats>(rg_reader->GetColumnStats(0));
+    // check values after serialization + deserialization
+    ASSERT_EQ(null_count, stats->null_count());
+    ASSERT_EQ(num_values - null_count, stats->num_values());
+    ASSERT_EQ(expected_stats.min(), stats->min());
+    ASSERT_EQ(expected_stats.max(), stats->max());
```

Comment:
Cool, as discussed on the PR the last thing that's missing here is correctness of the min and max stats